packages/yew/src/html/conversion.rs

```diff
@@ -79,17 +79,62 @@ impl_into_prop!(|value: &'static str| -> String { value.to_owned() });
 impl_into_prop!(|value: &'static str| -> Cow<'static, str> { Cow::Borrowed(value) });
 impl_into_prop!(|value: String| -> Cow<'static, str> { Cow::Owned(value) });
 
-/// A trait similar to `Into<Option<T>>` which allows conversion to an optional value of a
-/// `Properties` struct.
-pub trait IntoOptPropValue<T> {
-    /// Convert `self` to an optional value of a `Properties` struct.
-    fn into_opt_prop_value(self) -> Option<T>;
-}
-impl<T, V> IntoOptPropValue<V> for T
-where
-    T: IntoPropValue<Option<V>>,
-{
-    fn into_opt_prop_value(self) -> Option<V> {
-        self.into_prop_value()
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::{html::Properties, ComponentLink, Html};
+    use yew_macro::html;
+
+    #[derive(Properties, Clone)]
+    pub struct MockProps {
+        #[prop_or_default]
+        string: String,
+        #[prop_or_default]
+        string_opt: Option<String>,
+        #[prop_or_default]
+        cow_string: Cow<'static, str>,
+        #[prop_or_default]
+        cow_string_opt: Option<Cow<'static, str>>,
+        #[prop_or_default]
+        r#usize: usize,
+        #[prop_or_default]
+        usize_opt: Option<usize>,
+    }
+
+    pub struct MockComponent;
+
+    impl Component for MockComponent {
+        type Message = ();
+        type Properties = MockProps;
+
+        fn create(_props: Self::Properties, _link: ComponentLink<Self>) -> Self {
+            unimplemented!()
+        }
+
+        fn update(&mut self, _msg: Self::Message) -> bool {
+            unimplemented!()
+        }
+
+        fn change(&mut self, _props: Self::Properties) -> bool {
+            unimplemented!()
+        }
+
+        fn view(&self) -> Html {
+            unimplemented!()
+        }
+    }
+
+    #[test]
+    fn test_str() {
+        html! {
+            <MockComponent
+                cow_string="foo"
+                cow_string_opt="foo"
+                string="foo"
+                string_opt="foo"
+                r#usize=12
+                usize_opt=12
+                />
+        };
```

Comment:
Could you perhaps test the `IntoPropValue` macro directly instead of through the macro?
i.e. just:
```rust
IntoPropValue::<String>("foo");
IntoPropValue::<Option<String>>("foo");
```

Also, since these are essentially compile tests, we should consider using [trybuild](https://github.com/dtolnay/trybuild) for these too.

Reply:
I changed this to:

~~~rust
fn test_str() {
    let _: String = "foo".into_prop_value();
    let _: Option<String> = "foo".into_prop_value();
    let _: Cow<'static, str> = "foo".into_prop_value();
    let _: Option<Cow<'static, str>> = "foo".into_prop_value();
}
~~~